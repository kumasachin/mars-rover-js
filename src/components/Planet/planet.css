* {
    box-sizing: border-box;
}
.planet-container {
    text-align: center;
    padding: 20px;
}