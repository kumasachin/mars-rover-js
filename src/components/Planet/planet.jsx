import React from 'react';
import { useMarsContextAPI } from '../../context/marsContext';
import LABELS from '../../labels/';
import { Grid } from '../../modules/';
import './planet.css';

const Planet = () => {
  const {
    MESSAGE: { LOADING },
  } = LABELS;

  const data = useMarsContextAPI();

  const renderPlanet = () => {
    if (data.robots) {
      return <Grid />;
    }
    return <div>{LOADING}</div>;
  };

  return (
    <div className="planet-container">
      {data && data.robots && (
        <div className="planet-surface">{renderPlanet()}</div>
      )}
    </div>
  );
};

export default Planet;
