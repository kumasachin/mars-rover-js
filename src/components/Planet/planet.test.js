import { render, cleanup } from '@testing-library/react';
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Planet from './planet';

Enzyme.configure({ adapter: new Adapter() });

const data = {
  response: {
    map: {
      x: 1611,
      y: 16,
    },
    lostCell: {
      x: [],
      y: [],
    },
    robots: [
      {
        name: 'R1',
        color: 'green',
        currentPosition: '0 0 N',
        x: 0,
        y: 1,
        d: 'N',
        instructions: 'FFFFFFFFFFFFFFFFFFF',
      },
      {
        name: 'R2',
        color: 'green',
        currentPosition: '0 14 N',
        x: 0,
        y: 1,
        d: 'N',
        instructions: 'FRFFFFFFF',
      },
    ],
  },
};

jest.mock('../../context/marsContext', () => {
  return {
    useMarsContextAPI: () => data,
  };
});

describe('Planet Grid rendering', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Planet />);
    expect(asFragment(<Planet />)).toMatchSnapshot();
  });
});

describe('<Planet /> should', () => {
  let wrapper,
    createWrapper = (collection) =>
      mount(<Planet name="mars" data={collection} />);
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should have the button to start robot movement', async () => {
    wrapper = createWrapper(data.response.robots);
    expect(wrapper.find('button.init')).toBeDefined();
  });
});
