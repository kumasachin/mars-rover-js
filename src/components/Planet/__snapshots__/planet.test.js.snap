// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet Grid rendering should take a snapshot 1`] = `
<DocumentFragment>
  <h2>
    This is
  </h2>
  <button
    class="init"
    type="button"
  >
    Start moving robot
  </button>
  <div
    class="column"
  >
    <div
      class="console"
    >
      <header>
        <h1
          data-testid="application-heading"
        >
          Mars Rover project
        </h1>
      </header>
      <div
        class="consolebody"
      >
        <p>
          &gt;
        </p>
      </div>
    </div>
  </div>
  <div
    class="column "
  >
    <div>
      Click above to start
    </div>
  </div>
</DocumentFragment>
`;
